verde/utils.py

```diff
@@ -261,6 +261,137 @@ def maxabs(*args, nan=True, percentile=100):
+    >>> result = minmax((1, -10, 25, 2, 3, np.nan), nan=False)
+    >>> tuple(map(float, result))
+    (nan, nan)
+
+    If a more robust statistic is desired, you can use ``min_percentile`` and
+    or ``max_percentile`` to get the values at given percentiles instead of
+    the minimum and maximum.
+
+    >>> import numpy as np
+    >>> result = minmax(
+    ...     (1, -10, 25, 2, 3), min_percentile=2, max_percentile=98
+    ... )
+    >>> tuple(map(float, result))
+    (-9.12, 23.24)
+    >>> result = minmax(
+    ...     (1, -10, 25, 2, 3), min_percentile=0, max_percentile=100
+    ... )
+    >>> tuple(map(float, result))
+    (-10.0, 25.0)
+
+    """
+    arrays = [np.atleast_1d(i) for i in args]
+
+    # checks
+    if not isinstance(min_percentile, (float, int)):
+        raise TypeError(
+            f"Invalid 'min_percentile' of type '{type(min_percentile).__name__}'. Must be a float or int."
+        )
+    if not isinstance(max_percentile, (float, int)):
+        raise TypeError(
+            f"Invalid 'max_percentile' of type '{type(max_percentile).__name__}'. Must be a float or int."
+        )
+    if not min_percentile <= max_percentile:
+        raise ValueError(
+            f"'min_percentile' ({min_percentile}) must not be larger than 'max_percentile' ({max_percentile})."
+        )
+    if min_percentile < 0 or min_percentile > 100:
+        raise ValueError(
+            f"Invalid value for 'min_percentile' ({min_percentile}). Must be between 0 and 100."
+        )
+    if max_percentile < 0 or max_percentile > 100:
+        raise ValueError(
+            f"Invalid value for 'max_percentile' ({max_percentile}). Must be between 0 and 100."
+        )
+
+    # determine which functions to use
+    if nan:
+        npmin = np.nanmin
+        npmax = np.nanmax
+        nppercentile = np.nanpercentile
+    elif not nan:
+        npmin = np.min
+        npmax = np.max
+        nppercentile = np.percentile
+    else:
+        raise TypeError(f"The 'nan' parameter ({nan}) must be a boolean.")
+
+    # get min value
+    if min_percentile == 0:
+        min_ = npmin([npmin(i) for i in arrays])
+
+    # get max value
+    if max_percentile == 100:
+        max_ = npmax([npmax(i) for i in arrays])
+
+    # calculate min, max or both percentiles
+    if min_percentile != 0 or max_percentile != 100:
+        # concatenate values of all arrays
+        combined_array = np.concatenate([a.ravel() for a in arrays])
+
+        # if neither percentiles are defaults, calculate them together
+        if min_percentile != 0 and max_percentile != 100:
+            min_, max_ = nppercentile(combined_array, [min_percentile, max_percentile])
+
+        # only calculate min percentile
+        elif min_percentile != 0:
+            min_ = nppercentile(combined_array, min_percentile)
+
+        # only calculate max percentile
+        if max_percentile != 100:
+            max_ = nppercentile(combined_array, max_percentile)
+
+    return min_, max_
```

Comment:
I think we could simplify this logic a bit. What about this:
```suggestion
    if min_percentile == 0 and max_percentile == 0:
        min_ = npmin([npmin(i) for i in arrays])
        max_ = npmax([npmax(i) for i in arrays])
        return min_, max_

    # concatenate values of all arrays
    combined_array = np.concatenate([a.ravel() for a in arrays])
    min_, max_ = nppercentile(combined_array, [min_percentile, max_percentile])
    return min_, max_
```

If any of the `min_percentile` and `max_percentile` is different from the defaults, then we'll use `nppercentile` anyways. And a `min_percentile=0` will always lead to the minimum value. Moreover, I suspect there's no significant computational hit in passing two percentiles, since I guess that Numpy is computing each while looping over the elements in the array once.

So, we would only use `npmin` and `npmax` if `min_percentile==0` **and** `max_percentile==100`.

What do you think?

Reply:
There is a bit of time savings from only calculating one percentile and using min/max for the other. But maybe the time savings is not worth the extra code? At least that's how I interpreted the benchmarking results from below. The first uses min and max, with no percentiles, the second uses 1 and 99 percentiles, and the third uses min and 99 percentiles. As you can see, using 1 percentile and 1 min/max calculation is a bit fast than using 2 percentile calculations. 

What do you think?

```python
%timeit vd.minmax(a, min_percentile=0, max_percentile=100, nan=True)
%timeit vd.minmax(a, min_percentile=1, max_percentile=99, nan=True)
%timeit vd.minmax(a, min_percentile=0, max_percentile=99, nan=True)

>>>50.5 μs ± 2.56 μs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
>>>289 μs ± 26.5 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
>>>179 μs ± 3.93 μs per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```